server/src/events/handlers.ts

```diff
@@ -0,0 +1,70 @@
+import {promises as fs} from "fs";
+import path from "path";
+import {Server, Socket} from "socket.io";
+import {ChatService} from "../services/chat";
+import {MessageService} from "../services/message";
+import {Session_IF} from "../interfaces/session.if";
+import {DataSource} from "typeorm";
+import {Message_IF} from "../interfaces/message.if";
+import {File_If} from "../interfaces/file.if";
+
+export async function registerHandlers(io: Server, socket: Socket, session: Session_IF, db: DataSource) {
+    const chatService = new ChatService(db)
+    const msgService = new MessageService(db)
+    // const sessionKey = socket.handshake.auth.username
+    // if (sessionKey) {
+    //     await session.set(sessionKey, socket.id)
+    // }
+    //Client requesting to create a chat room with another client
+    socket.on("send chat", async (receiver: string, sender: string) => {
+        const chatId = await chatService.create(receiver, sender)
+        if (!chatId) {
+            return;
+        }
+        //Send to other client
+        const receiver_room = await session.get(receiver)
+        if (receiver_room) {
+            io.to(receiver_room).emit("accept chat", chatId)
+        }
+    })
+    //Get chat and a first bunch of messages
+    socket.on("get chat", async (chat: string, opts) => {
+        const response = await msgService.getSomeForChat(chat, opts)
+        socket.emit("send chat", response)
+    })
+    //Get all chats for a user
+    socket.on("get chats", async (user: string) => {
+        const chats = await chatService.getAll(user)
+        if (chats.length) {
+            socket.emit("send chats", chats)
+        }
+    })
+    //Client sending message to a room
+    socket.on("message", async (room: string, message: Message_IF) => {
+        //Log message to db
+        await msgService.post(message, room)
+        //Add users to the room if not already in (possible to send message to each sockets default room though, no real need to create one, but whatever)
+        if (!socket.rooms.has(room)) {
+            const users = await chatService.getUsers(room)
+            const sockets = await Promise.all(users.map(user => {
+                return {
+                    user: user, id: session.get(user)
+                }
+            }))
```

Comment:
```suggestion
            const sockets = await Promise.all(users.map(user => ({
                
                    user: user, id: session.get(user)
                
            })))
```
Чисто косметически, можшь не менять NB